storage-core/src/mmr/mod.rs

```diff
@@ -109,28 +201,46 @@ impl<H: CHasher> Mmr<H> {
         let mut mmr = Self {
             nodes: VecDeque::from(config.nodes),
             pruned_to_pos: config.pruned_to_pos,
-            pinned_nodes: HashMap::new(),
-            dirty_nodes: HashSet::new(),
+            pinned_nodes: BTreeMap::new(),
+            dirty_nodes: BTreeSet::new(),
             dirty_digest: Self::dirty_digest(),
+            #[cfg(feature = "std")]
             thread_pool: config.pool,
         };
         if mmr.size() == 0 {
             return mmr;
         }
 
-        for (i, pos) in Proof::<H::Digest>::nodes_to_pin(config.pruned_to_pos).enumerate() {
+        for (i, pos) in nodes_to_pin(config.pruned_to_pos).enumerate() {
             mmr.pinned_nodes.insert(pos, config.pinned_nodes[i]);
         }
 
         mmr
     }
 
+    /// Re-initialize the MMR with the given nodes, pruned_to_pos, and pinned_nodes.
+    pub fn re_init(
```

Comment:
nit: it might be more future-proof to have this take ownership of `self` and return `Self`. That way, if we add a new field to `Self`, we can't forget to set it here (or else it wouldn't compile.)

Reply:
done

Reply:
oops, actually this doesn't work, because I can't implement Copy in no_std